daweb/js2/sablony-komponenty/komponenty.md

```diff
@@ -0,0 +1,107 @@
+## Komponenty
+
+Abychom dokázali stavět větší aplikace a vyznali se v jejich kódu, bude se nám hodit rozdělit si vytváření obsahu stránky do menších funkcí. Představme si například hlavičku stránky s odkazy. Můžeme ji do stránky samozřejmě vložit takto:
+
+```jsx
+const appElement = document.querySelector('#app');
+appElement.innerHTML += render(
+  <header>
+    <a href="/">Domů</a>
+    <a href="/nakupy">Nákupy</a>
+    <a href="/kalendar">Kalendář</a>
+  </header>
+);
+```
+
+Do budoucna se nám však vyplatí vytvořit si pro tuto hlavičku samostatnou funkci. Nazveme ji `Header` a napíšeme ji tak, aby vracela JSX.
+
+```jsx
+const Header = () => {
+  return (
+    <header>
+      <a href="/">Domů</a>
+      <a href="/nakupy">Nákupy</a>
+      <a href="/kalendar">Kalendář</a>
+    </header>
+  );
+};
+```
+
+Tuto funkci pak můžeme použít při vytváření obsahu stránky.
+
+```jsx
+const appElement = document.querySelector('#app');
+appElement.innerHTML += render(Header());
+```
+
+## Skládání komponent
+
+Pojďme si pro cvik vytvořit ještě komponentu `Footer`, která bude představovat patičku stránky.
+
+```jsx
+const Footer = () => {
+  return (
+    <footer>
+      <p>© 2023 Czechitas, z. s.</p>
+    </footer>
+  );
+};
+```
+
+Kdybychom chtěli vytvořit kompletní stránku, mohli bychom ji napsat takto:
+
+```jsx
+const appElement = document.querySelector('#app');
+appElement.innerHTML += render(Header());
+appElement.innerHTML += render(
+  <main>
+    <h1>Ahoj ze světa JSX</h1>
+  </main>
+);
+appElement.innerHTML += render(Footer());
```

Comment:
Má nějaký hlubší záměr na pár řádcích jsx nepoužívat, používat, a pak zase nepoužívat?

Reply:
Mě to přijde jako ukázka toho, co ta funkce render dělá, je to jen takový rychlý mezikrok, hned za tím začneme skládat a tohle zapomeneme. Nepřijde mi to nějak zásadně matoucí.